Backend.ts

```diff
@@ -43,11 +43,24 @@ export class Backend<T> {
 			})
 		)
 	}
-	async list(document: { shard: string }): Promise<Document<T>[] | undefined> {
+	async list(
+		document: { shard: string },
+		limit?: number,
+		continuation?: string,
+		path?: string
+	): Promise<{ data: Document<T>[]; header: string | undefined }> {
```

Comment:
I think header should be called `continuation` or `continuationToken` to avoid associations with the "header object" that the `continuationToken` should be a part of.